packages/trpc/server/routers/viewer/eventTypes/getEventTypeOptions.handler.ts

```diff
@@ -0,0 +1,245 @@
+      name: profile.name,
+      image: getUserAvatarUrl({
+        avatarUrl: profile.avatarUrl,
+      }),
+      eventTypesLockedByOrg: parentOrgHasLockedEventTypes,
+    },
+    eventTypes: profileEventTypes,
+  });
+
+  eventTypeGroups = ([] as EventTypeGroup[]).concat(
+    eventTypeGroups,
+    await Promise.all(
+      memberships
+        .filter((mmship) => {
+          if (mmship?.team?.isOrganization) {
+            return false;
+          }
+          return true;
+        })
+        .map(async (membership) => {
+          const team = {
+            ...membership.team,
+            metadata: teamMetadataSchema.parse(membership.team.metadata),
+          };
+
+          const slug = null;
+
+          // if (forRoutingForms) {
+          //   // For Routing form we want to ensure that after migration of team to an org, the URL remains same for the team
+          //   // Once we solve this https://github.com/calcom/cal.com/issues/12399, we can remove this conditional change in slug
+          //   slug = `team/${team.slug}`;
+          // } else {
+          //   // In an Org, a team can be accessed without /team prefix as well as with /team prefix
+          //   slug = team.slug ? (!team.parentId ? `team/${team.slug}` : `${team.slug}`) : null;
+          // }
+
+          const eventTypes = team.eventTypes;
+          const teamParentMetadata = team.parent ? teamMetadataSchema.parse(team.parent.metadata) : null;
+          return {
+            teamId: team.id,
+            parentId: team.parentId,
+            bookerUrl: getBookerBaseUrlSync(team.parent?.slug ?? teamParentMetadata?.requestedSlug ?? null),
+            profile: {
+              name: team.name,
+              slug,
+            },
+            eventTypes: eventTypes
+              ?.filter((evType) => {
+                const res = evType.userId === null || evType.userId === user.id;
+                return res;
+              })
+              ?.filter((evType) =>
+                membership.role === MembershipRole.MEMBER
+                  ? evType.schedulingType !== SchedulingType.MANAGED
+                  : true
+              ),
+          };
+        })
+    )
+  );
+
+  let teamOptions: Option[] = [];
+
+  if (!skipTeamOptions) {
+    const profileTeamsOptions = eventTypeGroups
+      .map((group) => ({
+        ...group.profile,
+        teamId: group.teamId,
+      }))
+      .filter((profile) => !!profile.teamId)
+      .map((profile) => {
+        return {
+          value: String(profile.teamId) || "",
+          label: profile.name || profile.slug || "",
+        };
+      });
+
+    const otherTeams = await listOtherTeamHandler({ ctx });
+    const otherTeamsOptions = otherTeams
+      ? otherTeams.map((team) => {
+          return {
+            value: String(team.id) || "",
+            label: team.name || team.slug || "",
+          };
+        })
+      : [];
+
+    teamOptions = profileTeamsOptions.concat(otherTeamsOptions);
+  }
```

Comment:
moved the call to listOtherTeamHandler here instead of frontend because we only need it in case of org wide workflows and not normal or team workflows